### ISSUE TITLE
Add fluid states for form components

### ISSUE THREAD

**Issue by Akshat55** [carbon-design-system/carbon-components-angular#2769]
Majority of the form components have experimental support for fluid states.

- [ ] Combobox
- [ ] Date picker
- [ ] Dropdown
- [ ] Number Input
- [ ] Search
- [ ] Select
- [ ] Text Area
- [ ] Text Input/Password

![image](https://github.com/carbon-design-system/carbon-components-angular/assets/38994122/2aa9fa12-7f52-47ac-a240-1a2557c28c0b)

**Comment by Akshat55** [comment]
ref: https://github.com/carbon-design-system/carbon-components-angular/issues/2710

**Comment by kush-savani** [comment]
Hello @Akshat55, It this still open? I happy to help in this task.

**Comment by Akshat55** [comment]
Absolutely! Which component would you like me to assign you too? I prefer you approach this task one component at a time since fluid states are experimental for the time being.

**Comment by kush-savani** [comment]
hey @Akshat55, Assign me small one first :)

**Comment by Akshat55** [comment]
Assigned you to NumberInput @kush-savani.
